app/models/ssh_execution_provider.rb

```diff
@@ -23,6 +24,10 @@ def ssh_user(host)
       host.params['remote_execution_ssh_user']
     end
 
+    def ssh_port(host)
+      (host.params['remote_execution_ssh_port'] || Setting[:remote_execution_ssh_port]).to_i
```

Comment:
If the user accidentally enters a  string value this returns 0, which could be confusing/hard to troubleshoot.

Using something like this might be more desirable, so it raises if the user has entered something nonsensical:

```ruby
Integer(host.params['remote_execution_ssh_port'] || Setting[:remote_execution_ssh_port])
```